### DISCUSSION TITLE
How to load standard materials into a VRM Object

### DISCUSSION THREAD

**Discussion by aMidnightNova** [pixiv/three-vrm#1167] [Help] [answered]
It seams that Standard Materials are not being loaded in 1.0.7. 

How do I load standard materials now ?

here is a snapshot showing the difference in the returned materials property between version 0.6.7 and 1.0.7.

![image](https://user-images.githubusercontent.com/16892499/216017596-be514251-baa9-4421-bb20-ee8d3c5383d9.png)

**Comment by 0b5vr** [comment, marked as answer]
I believe standard materials are loaded properly by the `GLTFLoader`.

Starting from three-vrm v1.0, `VRM.materials` only has `MToonMaterial`s.
It's because materials need to be managed by the class `VRM` are only `MToonMaterial`s, which need to call `MToonMaterial.update()` upon `VRM.update()`.

You can use `gltf.parser.getDependencies('material')` instead, which lists every material loaded by the `GLTFLoader`.
The method can be applied not only for VRM but also for any glTF models.

![image](https://user-images.githubusercontent.com/7824814/216247002-487a5a86-7f29-4397-ac8c-435c7ad7cc22.png)

I apologize it's not documented on our [migration guide](https://github.com/pixiv/three-vrm/blob/dev/docs/migration-guide-1.0.md) properly.

**Reply by aMidnightNova** [reply]
Thanks for this, I will see if i can solve the issue I have now with this information.

I probably should of included it in the original post, but the TLDR of what lead me down this path is the model I am using is not being lit up by a directional light now, and I think its something to do with materials.

![image](https://user-images.githubusercontent.com/16892499/216257609-477b7e89-8647-40e7-a78b-451da8b76b71.png)

**Reply by aMidnightNova** [reply]
From your reply I was able to gather that three-vrm (0.6.7) was doing some sort of transforms to materials and after digging in the code for a bit I found that for Basic and Standard Materials three-vrm was doing a set of transforms on them as outlined below. Thank you for your reply, I don't think I would of figured that out as fast without it and fix the lighting issue I was running into.

```js
let convertGLTFMaterial = (material) => {
  if (material instanceof THREE.MeshStandardMaterial) {

    if (material.map) material.map.encoding = THREE.LinearEncoding
    if (material.emissiveMap) material.emissiveMap.encoding = THREE.LinearEncoding

    material.color.convertLinearToSRGB();
    material.emissive.convertLinearToSRGB();

  } else if (material instanceof THREE.MeshBasicMaterial) {

    if (material.map) material.map.encoding = THREE.LinearEncoding
    material.color.convertLinearToSRGB();

  }
}
```

**Reply by 0b5vr** [reply]
@aMidnightNova Oh! do you have `renderer.outputEncoding = THREE.sRGBEncoding` in your code?
It's recommended to set when you use GLTFLoader.
You should use the `renderer.outputEncoding` instead of modifying materials, to make it have a legit lighting.

See also the docs of Three.js GLTFLoader:
https://threejs.org/docs/#examples/en/loaders/GLTFLoader

**Reply by aMidnightNova** [reply]
Thanks for this, I found out about that flag at some point while trying to solve some Mixamo quaternion issues.
